### PR TITLE
Material field related fixes

### PR DESCRIPTION
**Pull request by Rexabit**
- Added more accurate renderer targets for applying materials to items.
- Fixed an issue where the material field could cause the screen to go black during icon creation.
- Added fallback object for icon snapshots (the ground item).

### CODE REVIEW

File: SetData/SetData.cs [hunk 1]
```diff
@@ -1033,18 +1033,18 @@ internal static void SetPieceRecipeData(PieceData data, ObjectDB Instant, bool c
                                         foreach (Renderer renderitem in renderfinder)
                                         {
                                             if (renderitem.receiveShadows && materialstr[0] != "none")
-                                                renderitem.material = mat;
```
Comment (Rexabit) [inline SetData/SetData.cs:1036]: Accessing 'material' will create a new instance of a material. So instead we're updating sharedMaterial.

File: Utility/functions.cs [hunk 2]
```diff
@@ -260,47 +252,62 @@ public static void SnapshotItem(ItemDrop item, float lightIntensity = 1.3f, Quat
...
+                Transform target = item.transform.Find("attach"); // ?? item.transform.Find("attach_skin");
 
-            camera.targetTexture = RenderTexture.GetTemporary((int)rect.width, (int)rect.height);
-            float maxDim = Mathf.Max(size.x, size.z);
-            float minDim = Mathf.Min(size.x, size.z);
-            float yDist = (maxDim + minDim) / Mathf.Sqrt(2) / Mathf.Tan(camera.fieldOfView * Mathf.Deg2Rad);
-            Transform transform = camera.transform;
-            transform.position = ((min + max) / 2) with { y = max.y } + new Vector3(0, yDist, 0);
-            topLight.transform.position = transform.position + new Vector3(-2, 0, 0.2f) / 3 * -yDist;
+                if (!target)
```
Comment (Rexabit) [inline Utility/functions.cs:261]: Added fallback game object in the event that 'attach' doesn't exist (in some cases it doesn't).

File: Utility/functions.cs [hunk 3]
```diff
@@ -260,47 +252,62 @@ public static void SnapshotItem(ItemDrop item, float lightIntensity = 1.3f, Quat
...
+                UnityEngine.Object.DestroyImmediate(visual);
+                camera.targetTexture.Release();
+            }
+            catch (System.Exception ex)
+            {
+                Debug.LogError($"[{WMRecipeCust.ModName}]: Failed to update icon - {ex.Message}");
+            }
+            finally
+            {
+                // Guarantee the deletion of these objects
```
Comment (Rexabit) [inline Utility/functions.cs:307]: Cleanup any leftover game objects in the event the function fails.

**Review by Wacky-Mole — APPROVED** [review]
Good Snapshot fixes

**Pull request merged**